astropy/coordinates/angles.py

```diff
@@ -587,7 +587,7 @@ def _validate_angles(self, angles=None):
         if angles.unit is u.deg:
             limit = 90
         elif angles.unit is u.rad:
-            limit = 0.5 * np.pi
+            limit = self.dtype.type(0.5 * np.pi)
```

Comment:
I am not sure if we can backport this to v5.0.x . This is the conflict I am seeing w.r.t. v5.0.x branch.

```
    def _validate_angles(self, angles=None):
        """Check that angles are between -90 and 90 degrees.
        If not given, the check is done on the object itself"""
        # Convert the lower and upper bounds to the "native" unit of
        # this angle.  This limits multiplication to two values,
        # rather than the N values in `self.value`.  Also, the
        # comparison is performed on raw arrays, rather than Quantity
        # objects, for speed.
        if angles is None:
            angles = self
<<<<<<< HEAD
        lower = u.degree.to(angles.unit, -90.0)
        upper = u.degree.to(angles.unit, 90.0)
=======

        if angles.unit is u.deg:
            limit = 90
        elif angles.unit is u.rad:
            limit = self.dtype.type(0.5 * np.pi)
        else:
            limit = u.degree.to(angles.unit, 90.0)

>>>>>>> a3a25b93d... Merge pull request #13745 from maxnoe/pihalf
```

Reply:
Same problem with v5.1.x

Reply:
Looks like https://github.com/astropy/astropy/pull/13498 needs to be backported too if we want this backported. If I don't hear back in a few days, I am going to change the milestone to v5.2 for this PR and not do any backports.

Reply:
Does it? Why not simply put `upper = self.dtype.type(u.degree.to(angles.unit, 90.0))` etc.?
Should not make a difference for `u.deg` as 90 is exact in both float32 and float 64, right?
And for any other units it should do no harm (might in fact have made sense to apply it to the `else` case here as well).

Reply:
Yes, that should be it, I can open the backport pr.